sway/commands/resize.c

```diff
@@ -76,105 +76,87 @@ void container_resize_tiled(struct sway_container *con,
 	}
 
 	// For HORIZONTAL or VERTICAL, we are growing in two directions so select
-	// both adjacent siblings. For RIGHT or DOWN, just select the next sibling.
-	// For LEFT or UP, convert it to a RIGHT or DOWN resize and reassign con to
-	// the previous sibling.
-	struct sway_container *prev = NULL;
-	struct sway_container *next = NULL;
+	// all adjacent siblings. For RIGHT or DOWN or LEFT or UP select just the
+	// previous or next sibling.
+	list_t *resize = create_list();
 	list_t *siblings = container_get_siblings(con);
 	int index = container_sibling_index(con);
 
 	if (axis == AXIS_HORIZONTAL || axis == AXIS_VERTICAL) {
-		if (index == 0) {
-			next = siblings->items[1];
-		} else if (index == siblings->length - 1) {
-			// Convert edge to top/left
-			next = con;
-			con = siblings->items[index - 1];
-			amount = -amount;
-		} else {
-			prev = siblings->items[index - 1];
-			next = siblings->items[index + 1];
-		}
+		list_cat(resize, siblings);
 	} else if (axis == WLR_EDGE_TOP || axis == WLR_EDGE_LEFT) {
 		if (!sway_assert(index > 0, "Didn't expect first child")) {
-			return;
+			goto cleanup;
 		}
-		next = con;
-		con = siblings->items[index - 1];
-		amount = -amount;
+		list_add(resize, siblings->items[index - 1]);
+		list_add(resize, con);
 	} else {
 		if (!sway_assert(index < siblings->length - 1,
 					"Didn't expect last child")) {
-			return;
+			goto cleanup;
 		}
-		next = siblings->items[index + 1];
+		list_add(resize, con);
+		list_add(resize, siblings->items[index + 1]);
 	}
 
 	// Apply new dimensions
-	int sibling_amount = prev ? ceil((double)amount / 2.0) : amount;
+	int sibling_amount = ceil((double)amount / (double)(resize->length - 1));
 
 	if (is_horizontal(axis)) {
-		if (con->pending.width + amount < MIN_SANE_W) {
-			return;
-		}
-		if (next->pending.width - sibling_amount < MIN_SANE_W) {
-			return;
-		}
-		if (prev && prev->pending.width - sibling_amount < MIN_SANE_W) {
-			return;
+		for(int i = 0; i < resize->length; i++) {
```

Comment:
Style nit: missing space between `for` and `(`

(Applies to other lines of code below as well.)